pretyping/typeclasses.ml

```diff
@@ -434,14 +434,23 @@ let remove_instance i =
   Lib.add_anonymous_leaf (instance_input (RemoveInstance, i));
   remove_instance_hint i.is_impl
 
-let declare_instance info local glob =
+let warning_not_a_class =
+  let name = "not-a-class" in
+  let category = "typeclasses" in
+  CWarnings.create ~name ~category (fun n ->
+      Pp.(str "Ignored instance declaration: "
+          ++ Constr.debug_print (printable_constr_of_global n)
```

Comment:
Use Nametab.shortest_qualid_of_global instead and print what it returns instead.

Reply:
Thus I have a similar question: how can I print a `types` (a.k.a. `Constr.t`)?

Reply:
Use Termops.Internal.print_constr_env